src/CellRenderer.ts

```diff
@@ -137,21 +141,26 @@ export class HeatCellRenderer extends ICellRenderer {
     this.installListener($cells);
 
     $cells
-      .style('background-color', (datum: any) => this.heatmapColorScale(datum))
+      .style('background-color', (datum: number) => this.heatmapColorScale(datum))
       .style('color', (datum: number) => adaptTextColorToBgColor(this.heatmapColorScale(datum).toString()))
       .text((d) => String(d));
 
     $cells
       .exit()
       .remove();
   }
+
+  clearCells() {
+    this.$parent.selectAll('div').remove();
+  }
 }
 
 export class ConfusionMatrixHeatCellRenderer extends HeatCellRenderer {
-  constructor(private cmdata: NumberMatrix, version1D: number[], private labels: [number, string]) {
-    super(version1D);
+  constructor(private cmdata: NumberMatrix, version1D: number[], private labels: [number, string], private $par: d3.Selection<any>) {
```

Comment:
Try to set `private $par: d3.Selection<any>` to protected in the child and parent class constructor.